src/core/types_.ml

```diff
@@ -1,35 +1,37 @@
-module TLS = Thread_local_storage_
+module TLS = Thread_local_storage
 module Domain_pool_ = Moonpool_dpool
 
-type ls_value = ..
-
-(** Key for task local storage *)
-module type LS_KEY = sig
-  type t
-  type ls_value += V of t
-
-  val offset : int
-  (** Unique offset *)
-
-  val init : unit -> t
-end
-
-type 'a ls_key = (module LS_KEY with type t = 'a)
-(** A LS key (task local storage) *)
-
 type task = unit -> unit
-type local_storage = ls_value array ref
+type fiber = Picos.Fiber.t
 
 type runner = {
-  run_async: ls:local_storage -> task -> unit;
+  run_async: fiber:fiber -> task -> unit;
   shutdown: wait:bool -> unit -> unit;
   size: unit -> int;
   num_tasks: unit -> int;
 }
 
 let k_cur_runner : runner TLS.t = TLS.create ()
-let k_cur_storage : local_storage TLS.t = TLS.create ()
-let _dummy_ls : local_storage = ref [||]
+[@@alert todo "remove me asap, done via picos now"]
+
+let k_cur_fiber : fiber TLS.t = TLS.create ()
+[@@alert todo "remove me asap, done via picos now"]
+
+let _dummy_computation : Picos.Computation.packed =
```

Comment:
I think optionals would be nicer than dummies and since there aren't too many in there, I think it's worth considering changing that.

Reply:
I agree, but optionals add one level of indirection :).